extensions/agroal/deployment/src/test/resources/application-default-datasource.properties

```diff
@@ -1,5 +1,5 @@
 #tag::basic[]
-quarkus.datasource.url=jdbc:h2:tcp://localhost/mem:default
+quarkus.datasource.url=jdbc:h2:mem:default
```

Comment:
What's the rationale of this change? I don't see how it is an improvement?

Reply:
I have read the thread.

Still don't see why we change all these URLs everywhere?

Reply:
Ah ok sorry, you are changing because in these tests you need more than one different DataSource so since now `H2DatabaseTestResource` can only provide one H2 (because the config is hardcoded) I cannot use it in these tests and I need to create an embedded memory database. This comes because of this limitation of hardcoding properties, then you cannot use it in these scenarions

Reply:
Yeah, frankly, considering this limitation, I don't see this as an improvement.

Reply:
Exactly this was what I have noticed after seeing the IT failing tests, but since I have already sent the PR I wanted to just discuss. So I think that for now, we can close it.